back-end/service.py

```diff
@@ -95,50 +131,58 @@ def updateSecret(secret, data, user, repo):
         log('debug', 'Secret updated in `{}`'.format(repo))
         return True
     else:
-        log('warning', 'Secret could not be updated in `{}`'.format(repo))
+        raise ServerError('Secret could not be updated in `{}`'.format(repo), 412)
         log('critical', 'Execution may not work properly from here.')
-        return False
 
 
 def ensureSecrets(user):
 
+    """
+    Runs a check on all of user's ACTIVATED Drone repositories
+    if each of them has the secret DRONE_PRIVATE_SSH_KEY and is updated
+    with the latest key.
+
+    Initiates the installation process for secret, if it doesn't exists.
+    """
+
     repos = requests.get(
         DRONE_ADDR + "/api/user/repos",
         headers={
             'Authorization': 'Bearer {}'.format(os.environ['DRONE_TOKEN'])
             }).json()
 
-    for repo in repos:
-        if repo['active']:
-
-            secrets = requests.get(
-                DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(user, repo['name']),
-                headers={'Authorization': 'Bearer {}'.format(
-                    os.environ['DRONE_TOKEN'])}
-                ).json()
+    for repo in [repo for repo in repos if repo['active']]:
 
-            with open(os.path.join(SSH_PATH, PRIV_KEY), 'r') as key:
+        secrets = requests.get(
+            DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(user, repo['name']),
+            headers={'Authorization': 'Bearer {}'.format(
+                os.environ['DRONE_TOKEN'])}
+            ).json()
 
-                for secret in secrets:
-                    if secret['name'] == 'DRONE_PRIVATE_SSH_KEY':
-                        log('debug', 'Secret found in repo `{}`'.format(
-                            repo['name']))
+        with open(os.path.join(SSH_PATH, PRIV_KEY), 'r') as key:
 
-                        return updateSecret(
-                            secret=secret['name'],
-                            data=key.read(),
-                            repo=repo['name'],
-                            user=user
-                        )
-                    else:
-                        return(writeSecret(key.read(), repo['name'], user))
+            for secret in secrets:
```

Comment:
Maybe I am missing something, but if the first `secret` in `secrets` does not satisfy `name`='DRONE_PRIVATE_SSH_KEY', then the loop will always immediately stop and return with `writeSecret()`, ignoring any other `secret` in `secrets`. Is this the expected behavior? Is a loop then actually necessary?

Reply:
I'd resolved this but for some reason it didn't get updated in the commit. Anyway, updated and pushed now. Please check.